scripts/auto-rebase/rebase.py

```diff
@@ -0,0 +1,317 @@
+
+    if len(matching_remote_refs) > 1:
+        matching_branches = ", ".join([r.name for r in matching_remote_refs])
+        logging.warning(f"Found more than one branch matching '{branch_name}' on remote: {matching_branches}. Taking first one")
+        _extra_msgs.append(f"Found more than one branch matching '{branch_name}' on remote: {matching_branches}.")
+        return matching_remote_refs[0]
+
+    if len(matching_remote_refs) == 1:
+        logging.info(f"Branch '{branch_name}' already exists on remote")
+        return matching_remote_refs[0]
+
+
+def is_local_branch_based_on_newer_base_branch_commit(git_repo, base_branch_name, remote_branch_name, local_branch_name):
+    """
+    Compares local and remote rebase branches by looking at their start on base branch.
+    Returns True if local branch is starts on newer commit and needs to be pushed to remote, otherwise False.
+    """
+    remote_merge_base = git_repo.merge_base(base_branch_name, remote_branch_name)
+    local_merge_base = git_repo.merge_base(base_branch_name, local_branch_name)
+
+    if remote_merge_base[0] == local_merge_base[0]:
+        logging.info(f"Remote branch is up to date. Branch-off commit: {commit_str(remote_merge_base[0])}")
+        return False
+    else:
+        logging.info(f"Remote branch is older - it needs updating. "
+                     f"Remote branch is on top of {base_branch_name}'s commit: '{commit_str(remote_merge_base[0])}'. "
+                     f"Local branch is on top of {base_branch_name}'s commit '{commit_str(local_merge_base[0])}'")
+        return True
+
+
+def try_get_pr(gh_repo, org, base_branch, branch_name):
+    prs = gh_repo.get_pulls(base=base_branch, head=f"{org}:{branch_name}", state="all")
+
+    if prs.totalCount == 0:
+        logging.info(f"PR for branch {branch_name} does not exist yet on {gh_repo.full_name}")
+        return None
+
+    pr = None
+    if prs.totalCount > 1:
+        pr = prs[0]
+        logging.warning(f"Found more than one PR for branch {branch_name} on {gh_repo.full_name} - this is unexpected, continuing with first one of: {[(x.state, x.html_url) for x in prs]}")
+
+    if prs.totalCount == 1:
+        pr = prs[0]
+        logging.info(f"Found PR #{pr.number} for branch {branch_name} on {gh_repo.full_name}: {pr.html_url}")
+
+    if pr.state == 'closed':
+        logging.warning(f"PR #{pr.number} is not open - new PR will be created")
+        if pr.is_merged():
+            logging.warning(f"PR #{pr.number} for '{branch_name}' branch is already merged but rebase.sh produced results")
+            _extra_msgs.append(f"PR #{pr.number} for '{branch_name}' was already merged but rebase.sh produced results")
+        else:
+            _extra_msgs.append(f"PR #{pr.number} for '{branch_name}' exists already but was closed")
+        return None
+    return pr
+
+
+def generate_pr_description(branch_name, amd_tag, arm_tag, prow_job_url, rebase_script_succeded):
+    base = (f"amd64: {amd_tag}\n"
```

Comment:
Maybe we can add the changelog contents from #1255 here, too, when that work is done.

Reply:
That's a good idea